src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -1713,12 +1786,17 @@ SubCmdValidate::execute(const SubCmdOptions& _options) const
   // Iterate through the test suites and compare them against the desired user tests
   // If a match is found enqueue the test suite to be executed
   for (size_t index = 0; index < testSuite.size(); ++index) {
+    std::string testSuiteName = get_test_name(testSuite[index].ptTest.get("name","<unknown>"));
     // The all option enqueues all test suites not marked explicit
     if (testsToRun[0] == "all") {
       // Do not queue test suites that must be explicitly passed in
       if(testSuite[index].ptTest.get<bool>("explicit"))
         continue;
       testObjectsToRun.push_back(&testSuite[index]);
+      // add custom param to the ptree if available
+      if (!param.empty() && param[0].compare(testSuiteName) == 0) {
```

Comment:
```suggestion
      if (!param.empty() && boost::equals(param[0], testSuiteName)) {
```
I think this is much cleaner than comparing to zero but that is just my opinion. I'm happy with either :)